### ISSUE TITLE
Google Analytics - Changes in region are captured in applicationWillResignActive

### ISSUE THREAD

**Issue by cagryInside** [OneBusAway/onebusaway-iphone#376]
While working on [OBA Android Google Analytics implementation](https://github.com/OneBusAway/onebusaway-android/issues/105), we realize that there is two different events for tracking region changes in the app. One where the user switches the region, and another when the region is automatically selected. 

The following code used for tracking the region switches within an action (```configured_region```):
```
- (void)applicationWillResignActive:(UIApplication *)application {
if([self.modelDao.readCustomApiUrl isEqualToString:@""]) {
[OBAAnalytics reportEventWithCategory:@"app_settings" action:@"configured_region" label:[NSString stringWithFormat:@"API Region: %@",self.modelDao.region.regionName] value:nil];
}else{
[OBAAnalytics reportEventWithCategory:@"app_settings" action:@"configured_region" label:@"API Region: Custom URL" value:nil];
}
```
However, in the IOS application lifecyle `applicationWillResignActive` will also be called when the user presses the home button or when the phone receives a call. So, this results lots of same events in the Google analytics console: 

![untitled 2](https://cloud.githubusercontent.com/assets/2777974/6449842/cbba1a26-c0f1-11e4-8fa4-797482f53c19.png)

As a solution, ```configured_region``` action could be called where the user switches the region.

**Comment by barbeau** [comment]
+1 for capturing this only when the user manually changes the region.  This metric doesn't really do us much good if its captured every time the user leaves the app.

**Comment by bbodenmiller** [comment]
Worth noting the OBA iOS analytics have a number of issues I hope to address this weekend.

**Comment by barbeau** [comment]
@bbodenmiller Ok, thanks.  Are there issues ticketed for those items yet?

**Comment by bbodenmiller** [comment]
Nothing I'm aware of other than the WIP PR: https://github.com/OneBusAway/onebusaway-iphone/pull/310

**Comment by barbeau** [comment]
To amend my comment above - as discussed in https://github.com/OneBusAway/onebusaway-android/issues/237#issuecomment-77021844, I didn't realize we were capturing these events both under ui_actions and app_settings.  I'm fine with these actions being recorded separately, but the "session" events definitely shouldn't be reported when the user leaves the app, since it results in a lot of extra events being reported (>= 2x-3x).  Looks like this problem is being addressed in #310.
